nexus/src/app/scim.rs

```diff
@@ -80,135 +72,279 @@ impl super::Nexus {
         silo_lookup: &lookup::Silo<'_>,
         token_id: Uuid,
     ) -> DeleteResult {
-        let (.., _authz_silo, _) =
-            silo_lookup.fetch_for(authz::Action::Delete).await?;
+        let (.., authz_silo, _) =
+            silo_lookup.fetch_for(authz::Action::ListChildren).await?;
+
+        self.datastore()
+            .scim_idp_delete_token_by_id(opctx, &authz_silo, token_id)
+            .await?;
 
-        let resource_type = ResourceType::ScimClientBearerToken;
-        let lookup_type = LookupType::by_id(token_id);
-        let not_found_error = lookup_type.into_not_found(resource_type);
-        let unimp = Unimpl::ProtectedLookup(not_found_error);
-        Err(self.unimplemented_todo(opctx, unimp).await)
+        Ok(())
     }
 
-    pub(crate) async fn scim_idp_delete_tokens_for_silo(
+    // SCIM client authentication
+
+    /// Authenticate a SCIM client based on a bearer token, and return a SCIM
+    /// provider store implementation that is scoped to a Silo.
+    pub(crate) async fn scim_idp_get_provider(
```

Comment:
I think it would be good for this name to make clearer that it handles token authentication. `scim_idp_get_provider` sounds like a lookup. It might even make sense to split it into two pieces, one that is the authn and one that gets the provider based on the token. But that's silly if they're always called together, and you don't really want to return the token back to the call site either. `scim_token_auth_and_get_provider` is a bit much I guess.

Reply:
This might be more relevant in the next PR, when you implement the actual SCIM operations, but it could make sense to mirror `op_context_for_external_api` and wrap up the authentication and the creation of the op context into one operation. Right now it looks like the unimplemented SCIM things don't take and OpContext but they will need to in order to do DB operations. Or at least, that's how I think it should work: op context created at top level at authn time and then passed into the provider methods.

https://github.com/oxidecomputer/omicron/blob/ef318b326abd25f7d3fd7efce4363cc07043d937/nexus/src/context.rs?plain=1#L335-L359